server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/fsWrite.ts

```diff
@@ -125,7 +129,15 @@ export class FsWrite {
     }
 
     public async requiresAcceptance(params: FsWriteParams): Promise<CommandValidation> {
-        return { requiresAcceptance: !(await this.workspace.getTextDocument(URI.file(params.path).toString())) }
+        // true when the file is not resolvable within our workspace. i.e. is outside of our workspace.
+        try {
+            const isInWorkspace = workspaceUtils.isInWorkspace(getWorkspaceFolderPaths(this.lsp), params.path)
```

Comment:
this looks like the same logic as that in `fsRead`, can we centralize it? Perhaps in toolShared?